program/src/instructions/token22/close_expired_listing.rs

```diff
@@ -28,6 +28,7 @@ pub struct CloseExpiredListingT22<'info> {
         bump = list_state.bump[0],
         close = rent_destination,
         has_one = owner,
+        constraint = list_state.expiry < Clock::get()?.unix_timestamp @ TcompError::ListingNotYetExpired,
```

Comment:
Same comment: should probably remove check in handler just to minimize overhead.